kpatch-build/kpatch-build

```diff
@@ -57,12 +57,14 @@ APPLIED_PATCHES=0
 OOT_MODULE=
 KLP_REPLACE=1
 
-GCC="${CROSS_COMPILE:-}gcc"
-CLANG="${CROSS_COMPILE:-}clang"
-LD="${CROSS_COMPILE:-}ld"
-LLD="${CROSS_COMPILE:-}ld.lld"
-READELF="${CROSS_COMPILE:-}readelf"
-OBJCOPY="${CROSS_COMPILE:-}objcopy"
+# default tools may be overridden by env
+: "${GCC:="${CROSS_COMPILE:-}gcc"}"
+: "${CLANG:="${CROSS_COMPILE:-}clang"}"
+: "${LD:="${CROSS_COMPILE:-}ld"}"
+: "${LLD:="${CROSS_COMPILE:-}ld.lld"}"
+: "${READELF:="${CROSS_COMPILE:-}readelf"}"
+: "${OBJCOPY:="${CROSS_COMPILE:-}objcopy"}"
+: "${MAKE:=make}"
 
```

Comment:
Yes, this is one of the changes for working with cross-compile, and using kernel-build-env designated toolchain, rather than the defaults a desktop build env might provide.

It's a layered default scheme, where CROSS_COMPILE bulk-overrides the tools that the default names might resolve from $PATH, but per-tool overrides can special-case that generic override.
Particularly useful (for example) where LLVM toolchain employed, but llvm's readelf didn't behave as required, and needed a wrapper script. Haven't documented that as I'm still unsure whether that's a local toolchain quirk, or a generic issue.

-: "${LLD:="${CROSS_COMPILE:-}ld.lld"}"
+# use CROSS_COMPILE as kernel/Makefile does, or the per-tool overrides if specified
+: set default to "${LLD:="${CROSS_COMPILE:-}ld.lld"}"
...

Yes, this also needs documenting in doc/INSTALL.md - none of the per-distro sections need mention these generalizations, for each distro will build its kernel with the tools its dev-environment presents in $PATH. But for cross-compiling, whether for a different arch in the same distro family, or for a wholly different scenario, these are useful.